File: lib/src/main/java/cc/lx/utils/Cocos2dxUtils.java
package cc.lx.utils;

/**
 * Cocos2dx 代理者
 */
public class Cocos2dxUtils {

    private static JsExecutor jsExecutor;

    public interface JsExecutor {
        void execute(String js);
    }

    public static void setJsExecutor(JsExecutor executor) {
        jsExecutor = executor;
    }

    public static void execute(String js) {
        if (jsExecutor != null) {
            jsExecutor.execute(js);
        }
    }
}

File: lib/src/main/java/com/xl/cocos/Cocos2dxUtils.kt
package com.xl.cocos

/**
 * Cocos2dx 代理者
 */
object Cocos2dxUtils {

    private var jsExecutor: Function1<String, Unit>? = null

    @JvmStatic
    fun setJsExecutor(executor: Function1<String, Unit>) {
        jsExecutor = executor
    }

    fun execute(js: String) {
        jsExecutor?.invoke(js)
    }
}